CMakeLists.txt

```diff
@@ -66,76 +34,135 @@ option(HIGHFIVE_HAS_WERROR "Convert warnings to errors." OFF)
 option(HIGHFIVE_HAS_FRIEND_DECLARATIONS "Enable additional friend declarations. Certain compiler require this On, others Off." OFF)
 mark_as_advanced(HIGHFIVE_HAS_FRIEND_DECLARATIONS)
 
-set(HIGHFIVE_SANITIZER OFF CACHE STRING "Enable a group of sanitizers, requires compiler support. Supported: 'address' and 'undefined'.")
-mark_as_advanced(HIGHFIVE_SANITIZER)
+option(HIGHFIVE_FIND_HDF5 "Find and link with HDF5." On)
+
+# Configure Tests & Examples
+# --------------------------
+
+# Internal variable that controls the default value for building
+# optional things like tests, examples and docs.
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(HIGHFIVE_EXTRAS_DEFAULT ON)
+else()
+  set(HIGHFIVE_EXTRAS_DEFAULT OFF)
+endif()
+
+option(HIGHFIVE_UNIT_TESTS "Compile unit-tests" ${HIGHFIVE_EXTRAS_DEFAULT})
+option(HIGHFIVE_EXAMPLES "Compile examples" ${HIGHFIVE_EXTRAS_DEFAULT})
+option(HIGHFIVE_BUILD_DOCS "Build documentation" ${HIGHFIVE_EXTRAS_DEFAULT})
 
-# In deployments we probably don't want/cant have dynamic dependencies
-option(HIGHFIVE_USE_INSTALL_DEPS "End applications by default use detected dependencies here" OFF)
-mark_as_advanced(HIGHFIVE_USE_INSTALL_DEPS)
+option(HIGHFIVE_TEST_BOOST "Enable Boost testing" OFF)
+option(HIGHFIVE_TEST_EIGEN "Enable Eigen testing" OFF)
+option(HIGHFIVE_TEST_OPENCV "Enable OpenCV testing" OFF)
+option(HIGHFIVE_TEST_XTENSOR "Enable xtensor testing" OFF)
+option(HIGHFIVE_TEST_HALF_FLOAT "Enable half-precision floats" OFF)
 
+# TODO remove entirely.
+option(HIGHFIVE_HAS_CONCEPTS "Print readable compiler errors w/ C++20 concepts" OFF)
+
+option(HIGHFIVE_HAS_WERROR "Convert warnings to errors." OFF)
+option(HIGHFIVE_GLIBCXX_ASSERTIONS "Enable bounds check for STL." OFF)
+# TODO these some magic to get a drop down menu in ccmake
+set(HIGHFIVE_SANITIZER OFF CACHE STRING "Enable a group of sanitizers, requires compiler support. Supported: 'address' and 'undefined'.")
+mark_as_advanced(HIGHFIVE_SANITIZER)
 
 # Check compiler cxx_std requirements
 # -----------------------------------
 
-if(CMAKE_CXX_STANDARD EQUAL 98)
-    message(FATAL_ERROR "HighFive needs to be compiled with at least C++11")
-endif()
-
 if(NOT DEFINED CMAKE_CXX_STANDARD)
-    set(CMAKE_CXX_STANDARD 11)
+    if(HIGHFIVE_TEST_XTENSOR)
+        set(CMAKE_CXX_STANDARD 14)
+    else()
+        set(CMAKE_CXX_STANDARD 11)
```

Comment:
Shouldn't this get updated now?

Reply:
That's been prepped in the other PR: #957.